File: frontend/src/models/Battle.ts

export type BattleSide =
    'team' |
    'enemy'

export type BattleWinLose =
    'win' |
    'lose' |
    'draw'

export type IkalampState =
    'live' |
    'death' |
    'sp' |
    'drop'

export type BattleRule =
    'nawabari' |
    'area' |
    'yagura' |
    'hoko' |
    'asari' |
    'unknown'

export type BattleStage = 
    'amabi' |
    'cyouzame' |
    'gonzui' |
    'hirame' |
    'kinmedai' |
    'konbu' |
    'kusaya' |
    'mahimahi' |
    'mantamaria' |
    'masaba' |
    'mategai' |
    'namerou' |
    'nampula' |
    'sumeshi' |
    'taraport' |
    'yagara' |
    'yunohana' |
    'zatou' |
    'takaashi' |
    'ohyou' |
    'bangaitei' |
    'negitoro' |
    'map' |
    'unknown'

export type MatchType = 
    'unknown' |
    'regular_match' |
    'bankara_match' |
    'x_match' |
    'event_match' |
    'fes_match' |
    'priv_match'

export type XMatchRate =
  'under_1500' |
  '1500_2000' |
  '2000_2500' |
  '2500_3000' |
  '3000_3500' |
  '3500_4000' |
  'upper_4000'

export type DeathReasonType = 
    'main_weapon' |
    'sub_weapon' |
    'sp_weapon' |
    'hoko_shoot' |
    'other' |
    'unknown'

export interface Ikalamp {
  side: BattleSide
  state: IkalampState
  bukiId: string
  xywh: number[]
  ord: number
  playImageWidth: number
  playImageHeight: number
  timestamp: number
  lowAccuracy: boolean
}

export interface WinLose {
    win: number
    lose: number
}

export interface BattleResult {
    rule: BattleRule
    stage: BattleStage
    result: WinLose
}

export const allRules: BattleRule[] = [
    'nawabari',
    'area',
    'yagura',
    'hoko',
    'asari'
]

export const allStages: BattleStage[] = [
    'amabi',
    'cyouzame',
    'gonzui',
    'hirame',
    'kinmedai',
    'konbu',
    'kusaya',
    'mahimahi',
    'mantamaria',
    'masaba',
    'mategai',
    'namerou',
    'nampula',
    'sumeshi',
    'taraport',
    'yagara',
    'yunohana',
    'zatou',
    'takaashi',
    'ohyou',
    'bangaitei',
    'negitoro'
]

export const allMatchTypes: MatchType[] = [
    'regular_match',
    'bankara_match',
    'x_match',
    'event_match',
    'fes_match',
    'priv_match'
]

export const allXMatchRates: XMatchRate[] = [
  'under_1500',
  '1500_2000',
  '2000_2500',
  '2500_3000',
  '3000_3500',
  '3500_4000',
  'upper_4000'
]

File: battle_analyzer/models/battle.py
from enum import Enum
from dataclasses import dataclass

class BattleSide(Enum):
    TEAM = 0
    ENEMY = 1
    NO_SIDE = 2

class BattleRule(Enum):
    UNKNOWN = 99
    NAWABARI = 0
    AREA = 1
    YAGURA = 2
    HOKO = 3
    ASARI = 4

class BattleStage(Enum):
    UNKNOWN = 99
    MAP = 100
    AMABI = 0
    CYOUZAME = 1
    GONZUI = 2
    HIRAME = 3
    KINMEDAI = 4 
    KONBU = 5
    KUSAYA = 6
    MAHIMAHI = 7
    MANTAMARIA = 8
    MASABA = 9
    MATEGAI = 10
    NAMEROU = 11
    NAMPULA = 12
    SUMESHI = 13
    TARAPORT = 14
    YAGARA = 15
    YUNOHANA = 16
    ZATOU = 17
    TAKAASHI = 18
    OHYOU = 19
    BANGAITEI = 20
    NEGITORO = 21
    KAZIKI = 22

class BattleWinLose(Enum):
    WIN = 0
    LOSE = 1
    DRAW = 3

class MatchType(Enum):
    UNKNOWN = 99
    REGULAR_MATCH = 0
    BANKARA_MATCH = 1
    X_MATCH = 2
    EVENT_MATCH = 3
    FES_MATCH = 4
    PRIV_MATCH = 5